### DISCUSSION TITLE
signup and login with username possible?

### DISCUSSION THREAD

**Discussion by aquatic7** [pallets-eco/flask-security#465] [Q&A]
Is it possible to do so email is not required if I want a user to signup and login with username instead?

**Comment by LonelyVikingMichael** [comment]
This functionality is well documented [here](https://flask-security-too.readthedocs.io/en/stable/configuration.html#SECURITY_USER_IDENTITY_ATTRIBUTES)

Your `User` model will require a `username` attribute, of course.
